packages/react-ui/src/components/molecules/UploadField.js

```diff
@@ -111,14 +114,26 @@ function UploadField(props) {
     setFilesText('');
   };
 
-  const dragPlaceholderText = dragOver ? props.dragPlaceholder : props.placeholder;
+  const setPlaceholder = () => {
+    const multipleSuffix = props.multiple ? '(s)' : '';
+    const placeholder = `Drag and drop your file${multipleSuffix} or click to browse`;
+    const dragPlaceholder = `Drop your file${multipleSuffix} here...`;
+
+    let placeholderText;
+    if (dragOver) {
+      placeholderText = dragPlaceholder;
+    } else {
+      placeholderText = props.placeholder || placeholder;
+    }
+    return placeholderText;
+  };
 
   return (
     <>
       <TextField
         {...props}
         ref={textFieldRef}
-        placeholder={dragPlaceholderText}
+        placeholder={setPlaceholder()}
```

Comment:
Benjamin is totally right, this is a getPlaceholder()

Reply:
It's already changed and using useMemo as he suggests 👍